### PR TITLE
Added integration of Slate doc format to Table View

### PR DESCRIPTION
**Pull request by RichardYSun**
Updates Table View to work with the Slate doc format. The redaction suggestion is bracketed right now (ie: "lah blah [redacted words] blah") so I will figure out a solution for that
This doesn't support users making their own redactions, but it can be stretch goal, because this view wasn't intended for making redactions.
image

### CODE REVIEW

File: frontend/src/assets/example_document.js [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
+                
             },
             {
                 text: " is in the business of early-stage health-care related research and product development. It alleges that Pfizer Canada ULC [Pfizer]",
-                [getMarkForCommentThreadID(overlappingCommentThreadID)]: true,
+                [getMarkForCommentThreadID(comments[1].id)]: true,
             },
             {
                 text: ", a major pharmaceutical company, infringes its patent by selling in Canada an auto-injector commonly known as the EpiPen. While, at first sight, the EpiPen and Seedlings’s invention do",
-                [getMarkForCommentThreadID(overlappingCommentThreadID)]: true,
```
Comment (wendi-yu) [inline frontend/src/assets/example_document.js:21]: just curious - why did you remove the slate fields here? Intentional?

File: frontend/src/components/document/DocumentView/StyledText.jsx [hunk 2]
```diff
@@ -1,32 +1,22 @@
 import { getCommentThreadsOnTextNode } from "@/util/editorCommentUtils";
+import { getRedactionsOnTextNode } from "@/util/editorRedactionUtils";
 import CommentedText from "./CommentedText";
+import RedactedText from "../Redactions/SuggestedRedaction";
```
Comment (wendi-yu) [inline frontend/src/components/document/DocumentView/StyledText.jsx:4]: let's use the `@/` notation :0 like `@/Redactions/SuggestedRedaction`?

File: frontend/src/components/document/DocumentView/StyledText.jsx [hunk 3]
```diff
@@ -1,32 +1,22 @@
...
+import RedactionPopover from "../Redactions/RedactionPopover";
+import {useState, useEffect, React} from "react";
 
 export default function StyledText({ attributes, children, leaf }) {
 
-  //TO DO: trim space off of highlight
```
Comment (wendi-yu) [inline frontend/src/components/document/DocumentView/StyledText.jsx:6]: Are these supposed to be removed? merge conflict? They seem kinda important lol

File: src/components/document/DocumentView/StyledText.jsx
Comment (wendi-yu) [inline src/components/document/DocumentView/StyledText.jsx:1]: empty file?

**Review by wendi-yu — COMMENTED** [review]
I'm kind of confused about the removals - are they supposed to be there or is this like a merge conflict with Emily's changes?

Also don't forget to lint ;)

File: src/components/document/DocumentView/StyledText.jsx
Comment (RichardYSun) [reply src/components/document/DocumentView/StyledText.jsx:1]: This might be from when I branched off of Emily's branch.

File: frontend/src/assets/example_document.js (hunk 1 above)
Comment (RichardYSun) [reply frontend/src/assets/example_document.js:21]: This was for testing, will bring it back

File: frontend/src/components/document/DocumentView/StyledText.jsx (hunk 3 above)
Comment (RichardYSun) [reply frontend/src/components/document/DocumentView/StyledText.jsx:6]: Yup, thanks for the catch

**Review by emilyylin — APPROVED** [review]
slay

File: frontend/src/util/editorRedactionUtils.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+function isRedactionIDMark(mayBeRedaction) {
+  return mayBeRedaction.indexOf(REDACTION_PREFIX) === 0;
+}
+
+export function getMarkForRedactionID(threadID) {
+  return `${REDACTION_PREFIX}${threadID}`;
+}
+
+export function insertRedaction(editor, addRedactionToState) {
```
Comment (emilyylin) [inline frontend/src/util/editorRedactionUtils.js:28]: i deleted these two functions from my recent PR

File: frontend/src/util/editorRedactionUtils.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const REDACTION_PREFIX = "redaction_";
+import { v4 as uuid } from "uuid";
```
Comment (emilyylin) [inline frontend/src/util/editorRedactionUtils.js:2]: and deleted these two lines bc unused imports

File: frontend/src/components/document/DocumentView/StyledText.jsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { getCommentThreadsOnTextNode } from "@/util/editorCommentUtils";
 import { getRedactionsOnTextNode, getMarkFromLeaf } from "@/util/editorRedactionUtils";
-import CommentedText from "@/components/document/Document/CommentedText";
+import CommentedText from "@/components/document/DocumentView/CommentedText";
```
Comment (RichardYSun) [inline frontend/src/components/document/DocumentView/StyledText.jsx:3]: This was giving the error: 
`Failed to load url ...`

Comment (wendi-yu) [reply frontend/src/components/document/DocumentView/StyledText.jsx:3]: woops ty for fix

File: frontend/src/components/document/TableView/TableView.jsx [hunk 7]
```diff
@@ -98,11 +80,42 @@ const TableEntry = (sentence) => {
...
+            }
+            curr_word_index += child.text.split(" ").length
+        })
+
+        return paragraph_index
+    })
+
+    var count = -1;
+    const tableEntries = []
+    paragraph_index_list.forEach((paragraph_index) => {
```
Comment (wendi-yu) [inline frontend/src/components/document/TableView/TableView.jsx:110]: you don't need the manual `count` here, you can get the idx in the `forEach` function
```js
myArray.forEach(function (value, i) {
    console.log('%d: %s', i, value);
});
```

**Review by wendi-yu — APPROVED** [review]
looks good :0
